src/backend/ml_api/routers/summarize.py

```diff
@@ -0,0 +1,25 @@
+import logging
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+from external_models.transformers_model import models
+from typing import Dict
+
+router = APIRouter()
+
+class SummarizeInputRequest(BaseModel):
+    text: str
+    model: str
+    
+class SummarizeOutputResponse(BaseModel):
+    response: str
+    
+@router.post("/")
+async def summarize(data: SummarizeInputRequest) -> SummarizeOutputResponse:
+    try:
+        model = models[data.model]
+        summary = model.generate_response(f"Summarize the following text: {data.text}")
```

Comment:
I'd suggest extracting prompts into external file or as a parameter, as some models might require very specific prompts to work well